ast/annotations.go

```diff
@@ -609,10 +610,11 @@ func (as *AnnotationSet) add(a *Annotations) *Error {
 		}
 	case annotationScopePackage:
 		if pkg, ok := a.node.(*Package); ok {
-			if exist, ok := as.byPackage[pkg]; ok {
+			hash := pkg.Path.Hash()
```

Comment:
For my understanding why do we hash the path here and not do something similar to the `annotationScopeSubpackages` case?

Reply:
For the `subpackages` scope, annotations are put in a tree structure, so they're accessible by all child nodes. The `package` scope should only be visible to rules within that package, but not sub-packages. The path of the package seems like the fitting lookup key, but sins it doesn't meet the equality criteria for being a map key, I'm hashing it. 
I'm in the process of reacquainting myself with golang and the project right now after being out of it for a bit, so if there is a more convenient/type safe way to accomplish this, I'm happy to take suggestions :).

Reply:
> The path of the package seems like the fitting lookup key, but sins it doesn't meet the equality criteria for being a map key, I'm hashing it.

Have you tried using `ast.Value` as the key and then using the package path? 

Also I'm trying to understand why the existing `pkg` key is no longer used. Will that result in wrong behavior if used?

Reply:
`ast.Value` doesn't meet the equality (`==`/`!=`) requirements to be a map key.

each module has its own `Package` instance, so using a `*Package` pointer as before gives you a unique key per module, regardless if their package namespaces overlap or not.

Reply:
Got it. Thanks for the clarification.